payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -0,0 +1,216 @@
+ *
+ * [GooglePayLauncherActivity] will:
+ * 1. Fetch the [StripeIntent]
+ * 2. Verify that Google Pay is available
+ * 3. Create a [PaymentDataRequest](https://developers.google.com/pay/api/web/reference/request-objects#PaymentDataRequest)
+ * 4. Confirm the [StripeIntent] using the Google Pay token
+ *
+ * Use [GooglePayLauncherContract] to start [GooglePayLauncherActivity].
+ *
+ * See [Troubleshooting](https://developers.google.com/pay/api/android/support/troubleshooting)
+ * for a guide to troubleshooting Google Pay issues.
+ */
+internal class GooglePayLauncherActivity : AppCompatActivity() {
+    private val paymentsClient: PaymentsClient by lazy {
+        PaymentsClientFactory(this).create(args.config.environment)
+    }
+    private val viewModel: GooglePayLauncherViewModel by viewModels {
+        GooglePayLauncherViewModel.Factory(
+            application,
+            args
+        )
+    }
+
+    private lateinit var args: GooglePayLauncherContract.Args
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (Build.VERSION.SDK_INT != Build.VERSION_CODES.O) {
+            // In Oreo, Activities where `android:windowIsTranslucent=true` can't request
+            // orientation. See https://stackoverflow.com/a/50832408/11103900
+            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        }
+
+        overridePendingTransition(0, 0)
+        setResult(
+            RESULT_OK,
+            Intent().putExtras(
+                GooglePayLauncherResult.Canceled.toBundle()
+            )
+        )
+
+        val nullableArgs = GooglePayLauncherContract.Args.fromIntent(intent)
+        if (nullableArgs == null) {
+            finishWithResult(
+                GooglePayLauncher.Result.Failed(
+                    RuntimeException(
+                        "GooglePayLauncherActivity was started without arguments."
+                    )
+                )
+            )
+            return
+        }
+        args = nullableArgs
+
+        viewModel.googlePayResult.observe(this) { googlePayResult ->
+            googlePayResult?.let(::finishWithResult)
+        }
+
+        if (!viewModel.hasLaunched) {
+            viewModel.hasLaunched = true
+
+            lifecycleScope.launch {
+                runCatching {
+                    viewModel.createPaymentDataRequest()
+                }.fold(
+                    onSuccess = ::isReadyToPay,
+                    onFailure = {
+                        viewModel.updateResult(
+                            GooglePayLauncher.Result.Failed(it)
+                        )
+                    }
+                )
+            }
+        }
+    }
+
+    override fun finish() {
+        super.finish()
+        overridePendingTransition(0, 0)
+    }
+
+    /**
+     * Check that Google Pay is available and ready
+     */
+    private fun isReadyToPay(paymentDataRequest: JSONObject) {
```

Comment:
I am curious why you chose not to use the DefaultGooglePayRepository

Reply:
Good point - I should be using that 😄